xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/ResourceReferenceRenamer.java

```diff
@@ -115,9 +146,18 @@ private boolean updateAbsoluteResourceReference(ResourceReference resourceRefere
             this.defaultReferenceDocumentReferenceResolver.resolve(linkEntityReference);
         EntityReference newTargetReference = newReference;
         ResourceType newResourceType = resourceReference.getType();
+        EntityReference absoluteResolvedReference = this.entityReferenceResolver.resolve(resourceReference, null);
 
-        // If the link targets the old (renamed) document reference, we must update it.
-        if (linkTargetDocumentReference.equals(oldReference)) {
+        // If the link targets the old (renamed) document reference and it's an absolute reference
+        // (i.e. its resolution without any given parameter gives same result than its resolution with the
+        // currentDocument) then we must update it
+        // We also update the link if it's not an absolute link but the current document is not part of the move job,
+        // as in this case there won't be any other call to perform the link refactoring.
+        boolean shouldBeUpdated = linkTargetDocumentReference.equals(oldReference)
+            && (absoluteResolvedReference.equals(linkEntityReference)
+            || !updatedEntities.containsKey(currentDocumentReference));
```

Comment:
This check doesn't work if the document with the link has already been moved before the current document I think.

Reply:
I think you're might be right on that one and I think it's the reason why part of my integration test isn't working see https://github.com/xwiki/xwiki-platform/pull/3553/files/7bb0a615e65405019dd416b286de29b5c7aa26c6#diff-223fd724fe6b367e09a02cb41234144641bea0082b58946fb457281668b7b092R716-R718

Now I felt it was better than nothing: it's not a regression for sure.